js-apps/meep-frontend/src/js/containers/idc-map.js

```diff
@@ -0,0 +1,425 @@
+import {
+  execChangeMap
+} from '../state/exec';
+import {
+  uiExecChangeSandboxCfg
+} from '../state/ui';
+
+class IDCMap extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.thisRef = createRef();
+    this.configRef = createRef();
+  }
+
+  componentDidMount() {
+    this.createMap();
+  }
+
+  componentWillUnmount() {
+    this.destroyMap();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.sandbox !== this.props.sandbox) {
+      this.destroyMap();
+      this.createMap();
+      this.updateMarkers();
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
+    // Check for map size update
+    let width = this.thisRef.current.offsetWidth;
+    let height = this.thisRef.current.offsetHeight;
+    if ((width && this.width !== width) || (height && this.height !== height)) {
+      this.width = width;
+      this.height = height;
+      // console.log('Map view resized to: ' + width + 'x' + height);
+      this.map.invalidateSize();
+      return true;
+    }
+
+    // Update if sandbox changed
+    if (nextProps.sandbox !== this.props.sandbox) {
+      return true;
+    }
+
+    // Update if map changed
+    if (!deepEqual(nextProps.execMap, this.props.execMap)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  createMap() {
+    // Get stored configuration
+    var cfg = this.props.sandboxCfg[this.props.sandboxName];
+    var lat = cfg.center ? cfg.center.lat : 43.73752;
+    var lng = cfg.center ? cfg.center.lng : 7.42892;
+    var zoom = cfg.zoom ? cfg.zoom : 15;
+    var baselayerName = cfg.baselayerName ? cfg.baselayerName : 'Positron';
+ 
+    // Create Map instance
+    var domNode = ReactDOM.findDOMNode(this);
+    this.map = L.map(domNode, {
+      center: [lat,lng],
+      zoom: zoom,
+      minZoom: 15,
+      maxZoom: 18,
+      drawControl: true
+    });
+
+    // Create GL Baselayers
+    var positronBaselayer = L.mapboxGL({style: 'http://10.3.16.105:30080/styles/positron/style.json'});
+    var darkBaselayer = L.mapboxGL({style: 'http://10.3.16.105:30080/styles/dark-matter/style.json'});
+    var klokBaselayer = L.mapboxGL({style: 'http://10.3.16.105:30080/styles/klokantech-basic/style.json'});
+    var osmBaselayer = L.mapboxGL({style: 'http://10.3.16.105:30080/styles/osm-bright/style.json'});
+    var baselayers = {
+      'Positron': positronBaselayer,
+      'Black Matter': darkBaselayer,
+      'Klokantech': klokBaselayer,
+      'OSM Bright': osmBaselayer
+    };
+
+    // Create Layer Group Overlays
+    this.ueOverlay = L.layerGroup();
+    this.uePathOverlay = L.layerGroup();
+    this.poaOverlay = L.layerGroup();
+    this.poaRangeOverlay = L.layerGroup();
+    this.computeOverlay = L.layerGroup();
+    var overlays = {
+      'UE': this.ueOverlay,
+      'UE Path': this.uePathOverlay,
+      'POA': this.poaOverlay,
+      'POA Range': this.poaRangeOverlay,
+      'Compute': this.computeOverlay
+    };
+    
```

Comment:
Instead of `UE` use `terminal`
Instead of `UE Path` use `terminal-path`
Instead pf `POA` use `poa`
Instead of `POA Range` use `poa-coverage`
Instead of `Compute` use `compute`

Reply:
Modified as suggested.